test/spark/app/src/main/scala/Sonnets.scala

```diff
@@ -41,6 +62,27 @@ object Sonnets {
         )
         target.save(outputPath)
 
+        {
+          /*
+           *  Verify all files match one of:
+           *  - s"${outputPath}/partition_key=PPP/XXXYYYZZZ.${fmt}"
+           * -  s"${outputPath}/_SUCCESS"
+           */
+          val pattern: Regex = s"${Regex.quote(outputPath)}/partition_key=[^/]*/.*/([^/]*\\.${Regex.quote(fmt)}|_SUCCESS)".r
+          val path = new Path(outputPath)
+          val fs = path.getFileSystem(sc.hadoopConfiguration)
+          val badFiles = convertToScalaIterator(fs.listFiles(path, true)).
+            map(file => s"${file.getPath.getParent}/${file.getPath.getName}").
+            filter(name => name match {
+              case pattern() => false
+              case _ => true
+            })
```

Comment:
I think you can keep pattern as string and just use something like 'filter(name => name match pattern)